p-token/src/processor/shared/transfer.rs

```diff
@@ -65,6 +65,9 @@ pub fn process_transfer(
     // remaining amount - the amount is only updated on the account if the transfer
     // is successful.
     let remaining_amount = if self_transfer {
+        // JC nit: the code in this branch gets executed for both branches, making
+        // this a bit harder to read. How about moving these lines outside of the
+        // the block, and only having an `if !self_transfer` branch?
```

Comment:
It uses more CU. 😊

Reply:
Understood, and yes, please comment in that case :smile: